installer/dev/tracelogging.h

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "pch.h"
+#include <wil/resource.h>
+#include <wil/result_macros.h>
+#include "..\..\dev\WindowsAppRuntime_Insights\WindowsAppRuntimeInsights.h"
+
+enum class InstallStage
+{
+    None                = 0x0,
+    InstallLicense      = 0x1,
+    GetPackageProperties= 0x2,
+    CreatePackageURI    = 0x3,
+    AddPackage          = 0x4,
+    RegisterPackage     = 0x5,
+    ProvisionPackage    = 0x6,
+};
+
+extern InstallStage g_installStage;
+extern PWSTR g_currentPackageFullName;
+extern HRESULT g_deploymentErrorExtendedHresult;
+extern PCWSTR g_deploymentErrorText;
+extern GUID g_deploymentErrorActivityId;
```

Comment:
globals and extern = blech

SUGGESTION: Create a class, make a static global instance, provide a function to get the instance and then share e.g. in InstallActivityContext.h

```
class InstallActivityContext
{
...
static InstallActivityContext& Get();

void Reset()
{
    m_foo = nullptr;
    m_bar = 0;
    ...
}
...
};
```

and InstallActivityContext.cpp

```
static InstallActivityContext g_installActivityContext;
InstallActivityContext& InstallActivityContext::Get()
{
   return g_installActivityContext;
}
```

Then callers can

```
#include <InstallActivityContext.h>
...
auto& context{ InstallActivityContext::Get() };
...
```